project-base/tests/ShopBundle/Database/Model/Product/ProductServiceTest.php

```diff
@@ -27,17 +31,24 @@ public function testRecalculateInputPriceForNewVatPercentWithInputPriceWithoutVa
 
         $vatData = new VatData('vat', 21);
         $vat = new Vat($vatData);
+        $availability = new Availability(new AvailabilityData([], 0));
 
         $pricingGroup = $this->getReference(PricingGroupDataFixture::PRICING_GROUP_ORDINARY_DOMAIN_1);
 
         $productData = new ProductData();
         $productData->price = 1000;
         $productData->vat = $vat;
         $productData->unit = $this->getReference(UnitDataFixture::UNIT_PIECES);
+        $productData->availability = $availability;
         $product = Product::create($productData);
 
         $productManualInputPrice = new ProductManualInputPrice($product, $pricingGroup, 1000);
 
+        $em->persist($vat);
```

Comment:
Why is it necessary to persist Vat and Product now? Why this was not necessary before your changes?